lib/model-utils.js

```diff
@@ -389,6 +389,8 @@ ModelUtils._coerce = function(where, options) {
         self._coerce(clauses[k], options);
       }
 
+      where[p] = clauses;
```

Comment:
This is kind of weird. Before my change, the block "handle logical operators" did not apply updates made by `coerceArray` to the original `where[p]` value. As a result, array-like object `{1: {field: 'value'}}` was not converted into an array `[{field: 'value'}]`. I am not sure why this was not caught earlier and surfaced only with Node.js 12. Is it perhaps a change in how deep-equality checks work on the new Node version? IDK.

Reply:
Here are the tests that are failing without my fix:

https://github.com/strongloop/loopback-datasource-juggler/blob/47addd943adfd7ace37cd52d2d4dcbd04ee7c0d8/test/loopback-dl.test.js#L1498-L1503